src/test/scala/outwatch/ScenarioTestSpec.scala

```diff
@@ -37,12 +38,18 @@ class ScenarioTestSpec extends JSDomSpec {
 
     document.getElementById("minus").dispatchEvent(event)
 
-    document.getElementById("counter").innerHTML shouldBe (-1).toString
-
-    for (i <- 0 to 10) {
-      document.getElementById("plus").dispatchEvent(event)
-      document.getElementById("counter").innerHTML shouldBe i.toString
-    }
+    Task {
```

Comment:
This whole block looks complicated compared to the old one. Is there a simpler way to write it?

Reply:
The simpler way would be to have the scheduler injected and use one with a synchronous execution model. Then the test code would be identical as before.

Reply:
Actually, the simplest way would be to have the synchronous scheduler everywhere (then it would be the same execution model as rxjs and the test could have stayed the same). But I don't like that, the batched execution model is a better choice to keep the page responsive.

Reply:
does the test fail with the batched execution model?

Reply:
Actually, it was not the batched execution model that was causing the async boundary here, it was the `Observable.merge` that uses a buffered subscriber to properly implement the back-pressure protocol. I've replaced it with a simple `mergeSync`, so the test stays unchanged.